### PR TITLE
Introduce `groupinds!` function to group parallel inds

### PR DESCRIPTION
**Pull request by mofeing**
CC @starsfordummies

### CODE REVIEW

**Review by starsfordummies — APPROVED** [review]
\o/

File: src/TensorNetwork.jl [hunk 1]
```diff
@@ -644,6 +647,38 @@ This method is equivalent to `tensor ∈ tensors(tn)` code, but it's faster on l
 Base.in(tensor::Tensor, tn::TensorNetwork) = tensor ∈ keys(tn.tensormap)
 Base.in(index::Symbol, tn::TensorNetwork) = index ∈ keys(tn.indexmap)
 
+"""
+    groupinds!(tn::AbstractTensorNetwork, i::Symbol)
+
+Group indices parallel to `i` and reshape the tensors to reflect it.
```
Comment (jofrevalles) [inline src/TensorNetwork.jl:653]:
```suggestion
Group indices parallel to `i` and reshape the tensors accordingly.
```

**Review by jofrevalles — CHANGES_REQUESTED** [review]

**Comment by jofrevalles** [comment]
Shouldn't we indicate which indices should the function group? Like `:i` and `:j`, and not only those who are "parallel to" `:i`?

**Comment by mofeing** [comment]
I see your point, but we are only going to "group" indices that are parallel; grouping indices that are not parallel involves performing contractions of the involved tensors, which we don't want to force.

**Pull request merged**